src/hooks/use-site-details.tsx

```diff
@@ -194,6 +195,7 @@ export function SiteDetailsProvider( { children }: SiteDetailsProviderProps ) {
 			siteName?: string,
 			wpVersion?: string,
 			customDomain?: string,
+			enableSSL?: boolean,
 			callback?: ( site: SiteDetails | void ) => Promise< void >
```

Comment:
```suggestion
			callback?: ( site: SiteDetails ) => Promise< void >
```
We only ever call the callback if we have a `SiteDetails` object. Let's take this opportunity to update the type signature. This will allow us to also simplify the callback in `src/hooks/use-add-site.ts`. 

I agree with your previous point that the data flows here are convoluted for no good reason and we should do more work on that. This is very low-hanging fruit, though.